dialog/file.go

```diff
@@ -30,10 +30,54 @@ type fileDialog struct {
 	win      *widget.PopUp
 	selected *fileDialogItem
 	callback interface{}
+	filter   FileFilter
 	dir      string
 	save     bool
 }
 
+// FileFilter is an interface that can be implemented to provide a filter to a file dialog
+type FileFilter interface {
+	Matches(fyne.URI) bool
+}
+
+type extensionFileFilter struct {
+	extensions []string
+}
+
+type mimeTypeFileFilter struct {
+	mimeTypes []string
+}
+
+// Matches returns true if a file URI has one of the filtered extensions
+func (e *extensionFileFilter) Matches(uri fyne.URI) bool {
+	extension := uri.Extension()
+	for _, ext := range e.extensions {
+		if extension == ext {
+			return true
+		}
+	}
+	return false
+}
+
+// Matches returns true if a file URI has one of the filtered mimetypes
+func (mt *mimeTypeFileFilter) Matches(uri fyne.URI) bool {
+	_, mimeType, mimeSubType := mimeTypeGet(uri)
+	for _, mimeTypeFull := range mt.mimeTypes {
+		mimeTypeSplit := strings.Split(mimeTypeFull, "/")
+		if len(mimeTypeSplit) <= 1 {
+			continue
+		}
+		mType := mimeTypeSplit[0]
+		mSubType := mimeTypeSplit[1]
```

Comment:
Should this strip ';*'?

Reply:
It already gets stripped by mimetype function

Reply:
What about mt.mimeTypes?

Reply:
Good point. Should probably strip in case the user provided mimetypes have it.